### DISCUSSION TITLE
Decimal.js precision config

### DISCUSSION THREAD

**Discussion by dstoyanoff** [prisma/prisma#20087] [Q&A] [answered]
Hello,
I have a question regarding the decimal.js integration.

By default, decimal.js uses a precision of 20. Since we also use decimal.js across our app, we use that instead of the prisma one. To increase precision, we have configured our Decimal for a precision of 30 instead of the default of 20 (https://mikemcl.github.io/decimal.js/#Dset). The question is how to do that on the Prisma decimal too? I have tried applying the same config to the Prisma.Decimal, which does work if I wrap the values in a `new Prisma.Decimal`, but the ones that arrived from the database are still using the old precision.

The only solution that I have for now is wrapping the values in a `new Decimal` manually, but this is easy to forget when querying the database.

In the below example test, I have tested using Prisma.Decimal and and directly using our configured Decimal

```
  test('t2', () => {
    const total = '85714.285710000000000000000000000000';
    const valueRaw = '0.472222222222222222222222222222';

    const value1 = new Prisma.Decimal(valueRaw).mul(new Prisma.Decimal(total)).div(100);
    const value2 = new Decimal(valueRaw).mul(new Decimal(total)).div(100);

    console.log(value1.toString(), value2.toString());
  });
```

This results in `404.76190474166666667 404.761904741666666666666666666`

**Comment by ludralph** [comment]
Hi @dstoyanoff 👋 

Can you please share how you are defining the field that stores the decimal value in your `prisma.schema` file?

for example, In the model below, 

```prisma
model Product {
  id Int @id @default(autoincrement())
  code String
  price Decimal @db.Decimal(9,2)
}
``` 
In the above code, the `price` field stores a Decimal value with a precision of 9 and scale of 2.

**Reply by dstoyanoff** [reply]
Hello @ludralph, I haven't configured a specific precision, so Prisma uses the postgres default of 65, 30

**Reply by ludralph** [reply]
Hi @dstoyanoff, thank you for providing the information. Since you haven't configured a specific precision for the Decimal type in Prisma, it uses the default precision which you mentioned. In order to align the precision with the configuration you have set for decimal.js (precision of 30), you can modify the Prisma schema with the precision you want so that subsequent data stored in the database can be stored in the defined precision without you having to do extra work to get it to the correct precision.

**Reply by dstoyanoff** [reply]
I am sorry, but I do not understand this how would that change things. The data is already stored properly in the database (you can see the example above). The issue is when returning it from the prisma client. The way decimal.js works is by setting it globally and not per instance, so I don't think doing that could affect in any way.

**Reply by ludralph** [reply]
I misunderstood the issue you're facing. It appears that the problem lies with how the Prisma client handles the precision of Decimal values when retrieving them from the database. One possible solution is to manually wrap the Decimal values returned by Prisma in a new Decimal instance with the desired precision. To avoid forgetting this step when querying the database, you could create a helper function or wrapper around the Prisma client's query methods. This wrapper would automatically apply the precision adjustment to the Decimal values before returning them.

**Reply by dstoyanoff** [reply]
Yes, this is what I do now. I will look into some automation with the extensions, thanks for the idea!

**Reply by ludralph** [reply]
Hi @dstoyanoff ,

Do you have any more questions around this issue? If no, it would be really helpful if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!

**Reply by sabinadams** [reply]
@dstoyanoff If you come up with an extension for this that works, please share here! I'd love to see it 😀

**Comment by MichalLytek** [comment, marked as answer]
Just do the same trick with `Prisma.Decimal` instead (e.g. in file where you export Prisma client):

```ts
Prisma.Decimal.set({
  precision: 30,
  toExpPos: 50,
});
```
